lib/python/rose/config.py

```diff
@@ -625,32 +1201,73 @@ def load_with_opts(self, source, node=None, more_keys=None,
+            more_keys (list - optional): A list of additional optional
+                configuration names. If source is "rose-${TYPE}.conf", the
+                file of each name should be "opt/rose-${TYPE}-${NAME}.conf".
+            used_keys (list - optional): If defined, it should be a list for
+                this method to  append to. The key of each successfully loaded
+                optional configuration will be appended to the list (unless the
+                key is already in the list). Missing optional configurations
+                that are specified in more_keys will not raise an error.
+                If not defined, any missing optional configuration will
+                trigger an OSError.
+            mark_opt_configs (bool - optional): if True, add comments above any
+                settings which have been loaded from an optional config.
+            return_config_map (bool - optional): If True, construct and return
+                a dict (config_map) containing config names vs their uncombined
+                nodes. Optional configurations use their opt keys as keys, and
+                the main configuration uses 'None'.
+
+        Returns:
+            tuple: node or (node, config_map):
+                - node - The loaded configuration as a ConfigNode.
+                - config_map - A dictionary containing opt_conf_key: ConfigNode
+                  pairs. Only returned if return_config_map is True.
+
+        Examples:
+            .. testcleanup:: rose.config.ConfigLoader.load_with_opts
+
+                try:
+                    os.remove('config.conf')
+                    os.remove('opt/config-foo.conf')
+                    os.rmdir('opt')
+                except OSError:
+                    pass
+
+            >>> # Write config file.
+            >>> with open('config.conf', 'w+') as config_file:
+            ...     config_file.write('''
+            ... [foo]
+            ... bar=Bar
+            ...     ''')
+            >>> # Write optional config file (foo).
+            >>> os.mkdir('opt')
+            >>> with open('opt/config-foo.conf', 'w+') as opt_config_file:
+            ...     opt_config_file.write('''
+            ... [foo]
+            ... bar=Baz
+            ...     ''')
+
+            >>> loader = ConfigLoader()
+            >>> config_node, config_map = loader.load_with_opts(
+            ...     'config.conf', more_keys=['foo'], return_config_map=True)
+            >>> config_node.get_value(keys=['foo', 'bar'])
+            'Baz'
+
+            >>> original_config_node = config_map[None]
+            >>> original_config_node.get_value(keys=['foo', 'bar'])
+            'Bar'
+
+            >>> optional_config_node = config_map['foo']
+            >>> optional_config_node.get_value(keys=['foo', 'bar'])
+            'Baz'
 
         """
-        node = self.load(source, node)
+        if not node:
+            node = self.load(source, node)
```

Comment:
Just noticed that this change will break loading of site/user global configuration. After this change, user global will never get loaded because `node` will be defined. Please revert.